### PR TITLE
Feature/load save text

### PR DESCRIPTION
**Pull request by janschopohl**

### CODE REVIEW

**Comment by janschopohl** [comment]
All the merge conflicts are probably there because gui-explosive-picnic was merged into this branch, I will try to fix this soon.

File: source/ui/main.qml [hunk 1]
```diff
@@ -82,18 +83,56 @@ ApplicationWindow {
     //this component is instantiated by the addProject method
     Component{
         id: projectComponent
-        Splitview{
-            anchors.fill: parent
+        Item {
+          anchors.fill: parent
+          Splitview{
```
Comment (goldsborough) [inline source/ui/main.qml]: spaces

File: source/ui/gui-project.cpp [hunk 2]
```diff
@@ -150,3 +204,9 @@ std::function<MemoryValue(std::string)> GuiProject::getUnsignedToMemoryValue() {
 std::function<MemoryValue(std::string)> GuiProject::getFloatToMemoryValue() {
   return floatToMemoryValue;
 }
+
+void GuiProject::_throwError(const std::string& message,
+                             const std::vector<std::string>& arguments) {
+  QString errorMessage = QString::fromStdString(message);
```
Comment (goldsborough) [inline source/ui/gui-project.cpp]: auto

File: source/ui/gui-project.cpp [hunk 3]
```diff
@@ -82,21 +110,47 @@ void GuiProject::reset() {
...
+  }
+}
+
+void GuiProject::saveSnapshot(QString qName) {
+  std::string name = qName.toStdString();
   // dont know, what to do
 }
 
-void GuiProject::loadSnapshot(QString name) {
+void GuiProject::loadSnapshot(QString qName) {
```
Comment (goldsborough) [inline source/ui/gui-project.cpp:161]: `const&`

File: source/ui/gui-project.cpp [hunk 4]
```diff
@@ -82,21 +110,47 @@ void GuiProject::reset() {
...
+  QString qName = path.path();
+  std::string filePath = qName.toStdString();
+  std::string text;
+  try {
+    text = Utility::loadFromFile(filePath);
+    QString qText = QString::fromStdString(text);
+    _editorComponent.setText(qText);
+  } catch (const std::exception& exception) {
+    _throwError(std::string("Could not load file!") + exception.what(),
+                std::vector<std::string>());
```
Comment (goldsborough) [inline source/ui/gui-project.cpp]: Does `{}` work here instead of the explicit vector and line break?

File: source/ui/gui-project.cpp [hunk 5]
```diff
@@ -82,21 +110,47 @@ void GuiProject::reset() {
...
+    emit saveTextAs();
+  } else {
+    saveTextAs(_defaultTextFileSavePath);
+  }
+}
+
+void GuiProject::saveTextAs(QUrl path) {
+  QString qName = path.path();
+  _defaultTextFileSavePath = qName;
+  std::string name = qName.toStdString();
```
Comment (goldsborough) [inline source/ui/gui-project.cpp]: auto auto auto

File: source/ui/gui-project.cpp [hunk 6]
```diff
@@ -34,6 +54,11 @@ GuiProject::GuiProject(QQmlContext* context,
         emit memoryChanged(address, length);
       });
 
+  _projectModule.getParserInterface().setThrowErrorCallback([this](
+      const std::string& message, const std::vector<std::string>& arguments) {
```
Comment (goldsborough) [inline source/ui/gui-project.cpp]: this is the same as the other PR? Make the lambda parameters `auto`

File: source/ui/gui-project.cpp [hunk 7]
```diff
@@ -34,6 +54,11 @@ GuiProject::GuiProject(QQmlContext* context,
         emit memoryChanged(address, length);
```
Comment (goldsborough) [inline source/ui/gui-project.cpp]: Don't we need `this->memoryChanged` for GCC here?

File: source/ui/editor-component.cpp [hunk 8]
```diff
@@ -95,39 +90,45 @@ EditorComponent::EditorComponent(QQmlContext *projectContext,
...
     QColor color;
-    if (error.severity() == CompileErrorSeverity::ERROR) {
-      color = QColor(Qt::red);
-    }
-    if (error.severity() == CompileErrorSeverity::WARNING) {
-      color = QColor(Qt::yellow);
-    }
-    if (error.severity() == CompileErrorSeverity::INFORMATION) {
-      color = QColor(Qt::blue);
+    switch (error.severity()) {
```
Comment (goldsborough) [inline source/ui/editor-component.cpp:129]: 👍

**Review by goldsborough — APPROVED** [review]

**Pull request merged**